src/main/java/graphql/execution/ExecutionContextBuilder.java

```diff
@@ -18,19 +18,77 @@
     private ValuesResolver valuesResolver;
     private Instrumentation instrumentation;
     private ExecutionId executionId;
+    private GraphQLSchema graphQLSchema;
+    private ExecutionStrategy queryStrategy;
+    private ExecutionStrategy mutationStrategy;
+    private ExecutionStrategy subscriptionStrategy;
+    private Object context;
+    private Object root;
+    private Document document;
+    private String operationName;
+    private Map<String, Object> variables;
 
-    public ExecutionContextBuilder(ValuesResolver valuesResolver, Instrumentation instrumentation) {
+    public ExecutionContextBuilder valuesResolver(final ValuesResolver valuesResolver) {
         this.valuesResolver = valuesResolver;
-        this.instrumentation = instrumentation;
+        return this;
     }
 
+    public ExecutionContextBuilder instrumentation(final Instrumentation instrumentation) {
```

Comment:
could you remove the final initialiser for the arguments?  also for the other methods? thanks